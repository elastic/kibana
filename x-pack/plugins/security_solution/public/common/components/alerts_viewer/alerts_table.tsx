/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import React, { useEffect, useMemo } from 'react';

import { Filter } from '../../../../../../../src/plugins/data/public';
import { TimelineIdLiteral } from '../../../../common/types/timeline';
import { StatefulEventsViewer } from '../events_viewer';
import { alertsDefaultModel } from './default_headers';
import { useManageTimeline } from '../../../timelines/components/manage_timeline';
import * as i18n from './translations';
export interface OwnProps {
  end: number;
  id: string;
  start: number;
}

const defaultAlertsFilters: Filter[] = [
  {
    meta: {
      alias: null,
      negate: false,
      disabled: false,
      type: 'phrase',
      key: 'event.kind',
      params: {
        query: 'alert',
      },
    },
    query: {
      bool: {
        filter: [
          {
            bool: {
              should: [
                {
                  match: {
                    'event.kind': 'alert',
                  },
                },
              ],
              minimum_should_match: 1,
            },
          },
        ],
      },
    },
  },
];

interface Props {
  timelineId: TimelineIdLiteral;
  endDate: number;
  startDate: number;
  pageFilters?: Filter[];
}

const AlertsTableComponent: React.FC<Props> = ({
  timelineId,
  endDate,
  startDate,
  pageFilters = [],
}) => {
  const alertsFilter = useMemo(() => [...defaultAlertsFilters, ...pageFilters], [pageFilters]);
  const { initializeTimeline } = useManageTimeline();

  useEffect(() => {
    initializeTimeline({
      id: timelineId,
      documentType: i18n.ALERTS_DOCUMENT_TYPE,
      footerText: i18n.TOTAL_COUNT_OF_ALERTS,
      title: i18n.ALERTS_TABLE_TITLE,
      unit: i18n.UNIT,
    });
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);
  return (
    <StatefulEventsViewer
      pageFilters={alertsFilter}
      defaultModel={alertsDefaultModel}
      end={endDate}
      id={timelineId}
      start={startDate}
    />
  );
};

export const AlertsTable = React.memo(AlertsTableComponent);
