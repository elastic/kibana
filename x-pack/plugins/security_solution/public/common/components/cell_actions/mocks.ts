/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

export const makeAction = (actionsName: string, icon: string = 'icon', order?: number) => ({
  id: actionsName,
  type: actionsName,
  order,
  getIconType: () => icon,
  getDisplayName: () => actionsName,
  getDisplayNameTooltip: () => actionsName,
  isCompatible: () => Promise.resolve(true),
  execute: () => {
    alert(actionsName);
    return Promise.resolve();
  },
});
