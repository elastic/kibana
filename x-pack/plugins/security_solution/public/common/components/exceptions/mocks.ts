/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import {
  Operator,
  ExceptionListItemSchema,
  ExceptionEntry,
  NestedExceptionEntry,
  FormattedEntry,
} from './types';
import { ExceptionList } from '../../../lists_plugin_deps';

export const getExceptionListMock = (): ExceptionList => ({
  id: '5b543420',
  created_at: '2020-04-23T00:19:13.289Z',
  created_by: 'user_name',
  list_id: 'test-exception',
  tie_breaker_id: '77fd1909-6786-428a-a671-30229a719c1f',
  updated_at: '2020-04-23T00:19:13.289Z',
  updated_by: 'user_name',
  namespace_type: 'single',
  name: '',
  description: 'This is a description',
  _tags: ['os:windows'],
  tags: [],
  type: 'endpoint',
  meta: {},
  totalItems: 0,
});

export const getExceptionItemEntryMock = (): ExceptionEntry => ({
  field: 'actingProcess.file.signer',
  type: 'match',
  operator: Operator.INCLUSION,
  value: 'Elastic, N.V.',
});

export const getNestedExceptionItemEntryMock = (): NestedExceptionEntry => ({
  field: 'actingProcess.file.signer',
  type: 'nested',
  entries: [{ ...getExceptionItemEntryMock() }],
});

export const getFormattedEntryMock = (isNested = false): FormattedEntry => ({
  fieldName: 'host.name',
  operator: 'is',
  value: 'some name',
  isNested,
});

export const getExceptionItemMock = (): ExceptionListItemSchema => ({
  id: 'uuid_here',
  item_id: 'item-id',
  created_at: '2020-04-23T00:19:13.289Z',
  created_by: 'user_name',
  list_id: 'test-exception',
  tie_breaker_id: '77fd1909-6786-428a-a671-30229a719c1f',
  updated_at: '2020-04-23T00:19:13.289Z',
  updated_by: 'user_name',
  namespace_type: 'single',
  name: '',
  description: 'This is a description',
  comments: [
    {
      created_by: 'user_name',
      created_at: '2020-04-23T00:19:13.289Z',
      comment: 'Comment goes here',
    },
  ],
  _tags: ['os:windows'],
  tags: [],
  type: 'simple',
  entries: [
    {
      field: 'actingProcess.file.signer',
      type: 'match',
      operator: Operator.INCLUSION,
      value: 'Elastic, N.V.',
    },
    {
      field: 'host.name',
      type: 'match',
      operator: Operator.EXCLUSION,
      value: 'Global Signer',
    },
    {
      field: 'file.signature',
      type: 'nested',
      entries: [
        {
          field: 'signer',
          type: 'match',
          operator: Operator.INCLUSION,
          value: 'Evil',
        },
        {
          field: 'trusted',
          type: 'match',
          operator: Operator.INCLUSION,
          value: 'true',
        },
      ],
    },
  ],
});
