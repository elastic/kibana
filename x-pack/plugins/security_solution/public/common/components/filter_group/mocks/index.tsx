/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

import type { ControlGroupContainer } from '@kbn/controls-plugin/public';
import type { Filter } from '@kbn/es-query';
import type { FC } from 'react';
import React, { useEffect } from 'react';
import { TEST_IDS } from '../constants';
import type { FilterGroupProps } from '../types';
import { getControlGroupMock } from './control_group';

/**
 *
 *  Retrieves a simple of FilterGroup Item.
 *   - Can also call callback onFilterChange with custom args
 *
 * @example
 * const onFilterchangeMock = jest.fn();
 * const onInitMock = jest.fn();
 *
 *const TestComponent = () => (
 *  <TestProviders>
 *    <FilterGroup
 *      chainingSystem="NONE"
 *      dataViewId=""
 *      initialControls={[]}
 *      onFilterChange={onFilterchangeMock}
 *      onInit={onInitMock}
 *    />
 *  </TestProviders>
 *);
 *
 *jest.mock('..');
 *
 *describe('Some test', () => {
 *  it('basic test', () => {
 *    (FilterGroup as jest.Mock).mockImplementationOnce(
 *      getMockedFilterGroupWithCustomFilters([
 *        {
 *          meta: {
 *            params: ['open'],
 *          },
 *        },
 *      ])
 *    );
 *
 *    render(<TestComponent />);
 *
 *    expect(onFilterchangeMock.mock.calls[0][0]).toMatchObject([
 *      {
 *        meta: {
 *          params: ['open'],
 *        },
 *      },
 *    ]);
 *  });
 *});
 *
 */
export function getMockedFilterGroupWithCustomFilters(outputFilters?: Filter[]) {
  const FilterGroup: FC<FilterGroupProps> = ({ onInit, onFilterChange }) => {
    useEffect(() => {
      if (onInit) {
        onInit(getControlGroupMock() as unknown as ControlGroupContainer);
      }

      if (onFilterChange) {
        onFilterChange(outputFilters ?? []);
      }
    }, [onInit, onFilterChange]);

    return <div data-test-subj={TEST_IDS.MOCKED_CONTROL} />;
  };

  return FilterGroup;
}
