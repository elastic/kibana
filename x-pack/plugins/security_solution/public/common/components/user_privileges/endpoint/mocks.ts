/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

import type { EndpointPrivileges } from './use_endpoint_privileges';
import { getEndpointPrivilegesInitialState } from './utils';

export const getEndpointPrivilegesInitialStateMock = (
  overrides: Partial<EndpointPrivileges> = {}
): EndpointPrivileges => {
  // Get the initial state and set all permissions to `true` (enabled) for testing
  const endpointPrivilegesMock: EndpointPrivileges = {
    ...(
      Object.entries(getEndpointPrivilegesInitialState()) as Array<
        [keyof EndpointPrivileges, boolean]
      >
    ).reduce((mockPrivileges, [key, value]) => {
      mockPrivileges[key] = !value;

      return mockPrivileges;
    }, {} as EndpointPrivileges),
    ...overrides,
  };

  return endpointPrivilegesMock;
};
