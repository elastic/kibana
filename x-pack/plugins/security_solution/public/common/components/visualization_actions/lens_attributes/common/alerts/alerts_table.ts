/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */
import { v4 as uuidv4 } from 'uuid';
import type { GetLensAttributes } from '../../../types';

const layerId = uuidv4();

export const getAlertsTableLensAttributes: GetLensAttributes = (
  stackByField = 'kibana.alert.rule.name',
  extraOptions
) => {
  return {
    title: 'Alerts',
    description: '',
    visualizationType: 'lnsDatatable',
    state: {
      visualization: {
        columns: [
          {
            columnId: '2881fedd-54b7-42ba-8c97-5175dec86166',
            isTransposed: false,
            width: 362,
          },
          {
            columnId: 'f04a71a3-399f-4d32-9efc-8a005e989991',
            isTransposed: false,
          },
          {
            columnId: '75ce269b-ee9c-4c7d-a14e-9226ba0fe059',
            isTransposed: false,
          },
        ],
        layerId,
        layerType: 'data',
      },
      query: {
        query: '',
        language: 'kuery',
      },
      filters: extraOptions?.filters ? extraOptions.filters : [],
      datasourceStates: {
        formBased: {
          layers: {
            [layerId]: {
              columns: {
                '2881fedd-54b7-42ba-8c97-5175dec86166': {
                  label: `Top values of ${stackByField}`,
                  dataType: 'string',
                  operationType: 'terms',
                  scale: 'ordinal',
                  sourceField: stackByField,
                  isBucketed: true,
                  params: {
                    size: 1000,
                    orderBy: {
                      type: 'column',
                      columnId: 'f04a71a3-399f-4d32-9efc-8a005e989991',
                    },
                    orderDirection: 'desc',
                    otherBucket: true,
                    missingBucket: false,
                    parentFormat: {
                      id: 'terms',
                    },
                    include: [],
                    exclude: [],
                    includeIsRegex: false,
                    excludeIsRegex: false,
                  },
                },
                'f04a71a3-399f-4d32-9efc-8a005e989991': {
                  label: `Count of ${extraOptions?.breakdownField}`,
                  dataType: 'number',
                  operationType: 'count',
                  isBucketed: false,
                  scale: 'ratio',
                  sourceField: extraOptions?.breakdownField,
                  params: {
                    emptyAsNull: true,
                  },
                },
                '75ce269b-ee9c-4c7d-a14e-9226ba0fe059': {
                  label: `Top values of ${extraOptions?.breakdownField}`,
                  dataType: 'string',
                  operationType: 'terms',
                  scale: 'ordinal',
                  sourceField: extraOptions?.breakdownField,
                  isBucketed: true,
                  params: {
                    size: 1000,
                    orderBy: {
                      type: 'column',
                      columnId: 'f04a71a3-399f-4d32-9efc-8a005e989991',
                    },
                    orderDirection: 'desc',
                    otherBucket: true,
                    missingBucket: false,
                    parentFormat: {
                      id: 'terms',
                    },
                    include: [],
                    exclude: [],
                    includeIsRegex: false,
                    excludeIsRegex: false,
                  },
                },
              },
              columnOrder: [
                '2881fedd-54b7-42ba-8c97-5175dec86166',
                '75ce269b-ee9c-4c7d-a14e-9226ba0fe059',
                'f04a71a3-399f-4d32-9efc-8a005e989991',
              ],
              sampling: 1,
              incompleteColumns: {},
            },
          },
        },
        textBased: {
          layers: {},
        },
      },
      internalReferences: [],
      adHocDataViews: {},
    },
    references: [
      {
        type: 'index-pattern',
        id: '{dataViewId}',
        name: `indexpattern-datasource-layer-${layerId}`,
      },
    ],
  };
};
