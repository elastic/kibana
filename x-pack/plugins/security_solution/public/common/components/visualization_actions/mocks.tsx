/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */
import React from 'react';
import { cloneDeep } from 'lodash/fp';

import {
  TestProviders,
  mockGlobalState,
  SUB_PLUGINS_REDUCER,
  kibanaObservable,
  createSecuritySolutionStorageMock,
} from '../../mock';
import type { State } from '../../store';
import { createStore } from '../../store';

export const queryFromSearchBar = {
  query: 'host.name: *',
  language: 'kql',
};

export const filterFromSearchBar = [
  {
    meta: {
      alias: null,
      negate: false,
      disabled: false,
      type: 'phrase',
      key: 'host.id',
      params: {
        query: '123',
      },
    },
    query: {
      match_phrase: {
        'host.id': '123',
      },
    },
  },
];

export const mockCreateStoreWithQueryFilters = () => {
  const { storage } = createSecuritySolutionStorageMock();

  const state: State = mockGlobalState;

  const myState = cloneDeep(state);

  myState.inputs = {
    ...myState.inputs,
    global: {
      ...myState.inputs.global,
      query: queryFromSearchBar,
      filters: filterFromSearchBar,
    },
  };
  return createStore(myState, SUB_PLUGINS_REDUCER, kibanaObservable, storage);
};

export const wrapper = ({ children }: { children: React.ReactElement }) => (
  <TestProviders store={mockCreateStoreWithQueryFilters()}>{children}</TestProviders>
);
