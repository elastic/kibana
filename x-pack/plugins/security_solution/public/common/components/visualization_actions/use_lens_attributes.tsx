/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

import { useMemo } from 'react';
import { SecurityPageName } from '../../../../common/constants';
import { HostsTableType } from '../../../hosts/store/model';
import { NetworkRouteType } from '../../../network/pages/navigation/types';
import { useSourcererDataView } from '../../containers/sourcerer';
import { useDeepEqualSelector } from '../../hooks/use_selector';
import { inputsSelectors } from '../../store';
import { useRouteSpy } from '../../utils/route/use_route_spy';
import type { LensAttributes, GetLensAttributes } from './types';
import {
  getHostDetailsPageFilter,
  sourceOrDestinationIpExistsFilter,
  hostNameExistsFilter,
  getIndexFilters,
} from './utils';

export const useLensAttributes = ({
  lensAttributes,
  getLensAttributes,
  stackByField,
  title,
}: {
  lensAttributes?: LensAttributes | null;
  getLensAttributes?: GetLensAttributes;
  stackByField?: string;
  title?: string;
}): LensAttributes | null => {
  const { selectedPatterns, dataViewId } = useSourcererDataView();
  const getGlobalQuerySelector = useMemo(() => inputsSelectors.globalQuerySelector(), []);
  const getGlobalFiltersQuerySelector = useMemo(
    () => inputsSelectors.globalFiltersQuerySelector(),
    []
  );
  const query = useDeepEqualSelector(getGlobalQuerySelector);
  const filters = useDeepEqualSelector(getGlobalFiltersQuerySelector);
  const [{ detailName, pageName, tabName }] = useRouteSpy();

  const tabsFilters = useMemo(() => {
    if (pageName === SecurityPageName.hosts && tabName === HostsTableType.events) {
      return hostNameExistsFilter;
    }

    if (pageName === SecurityPageName.network && tabName === NetworkRouteType.events) {
      return sourceOrDestinationIpExistsFilter;
    }

    return [];
  }, [pageName, tabName]);

  const pageFilters = useMemo(() => {
    if (pageName === SecurityPageName.hosts && detailName != null) {
      return getHostDetailsPageFilter(detailName);
    }
    return [];
  }, [detailName, pageName]);

  const indexFilters = useMemo(() => getIndexFilters(selectedPatterns), [selectedPatterns]);

  const lensAttrsWithInjectedData = useMemo(() => {
    if (lensAttributes == null && (getLensAttributes == null || stackByField == null)) {
      return null;
    }
    const attrs: LensAttributes =
      lensAttributes ??
      ((getLensAttributes && stackByField && getLensAttributes(stackByField)) as LensAttributes);

    return {
      ...attrs,
      ...(title != null ? { title } : {}),
      state: {
        ...attrs.state,
        query,
        filters: [
          ...attrs.state.filters,
          ...filters,
          ...pageFilters,
          ...tabsFilters,
          ...indexFilters,
        ],
      },
      references: attrs.references.map((ref: { id: string; name: string; type: string }) => ({
        ...ref,
        id: dataViewId,
      })),
    } as LensAttributes;
  }, [
    lensAttributes,
    getLensAttributes,
    stackByField,
    title,
    query,
    filters,
    pageFilters,
    tabsFilters,
    indexFilters,
    dataViewId,
  ]);

  return lensAttrsWithInjectedData;
};
