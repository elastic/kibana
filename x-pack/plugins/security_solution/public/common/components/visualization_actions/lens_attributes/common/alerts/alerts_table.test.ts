/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */
import { renderHook } from '@testing-library/react-hooks';
import { wrapper } from '../../../mocks';

import { useLensAttributes } from '../../../use_lens_attributes';

import { getAlertsTableLensAttributes } from './alerts_table';

jest.mock('uuid', () => ({
  v4: jest.fn().mockReturnValue('4aa7cf71-cf20-4e62-8ca6-ca6be6b0988b'),
}));

jest.mock('../../../../../containers/sourcerer', () => ({
  useSourcererDataView: jest.fn().mockReturnValue({
    dataViewId: 'security-solution-my-test',
    indicesExist: true,
    selectedPatterns: ['signal-index'],
  }),
}));

jest.mock('../../../../../utils/route/use_route_spy', () => ({
  useRouteSpy: jest.fn().mockReturnValue([
    {
      pageName: 'alerts',
    },
  ]),
}));

describe('getAlertsTableLensAttributes', () => {
  it('should render without extra options', () => {
    const { result } = renderHook(
      () =>
        useLensAttributes({
          getLensAttributes: getAlertsTableLensAttributes,
          stackByField: 'event.category',
        }),
      { wrapper }
    );

    expect(result?.current).toMatchSnapshot();
  });

  it('should render with extra options - filters', () => {
    const { result } = renderHook(
      () =>
        useLensAttributes({
          extraOptions: {
            filters: [
              {
                meta: {
                  type: 'phrases',
                  key: '_index',
                  params: ['.alerts-security.alerts-default'],
                  alias: null,
                  negate: false,
                  disabled: false,
                },
                query: {
                  bool: {
                    should: [
                      {
                        match_phrase: {
                          _index: '.alerts-security.alerts-default',
                        },
                      },
                    ],
                    minimum_should_match: 1,
                  },
                },
              },
            ],
          },
          getLensAttributes: getAlertsTableLensAttributes,
          stackByField: 'event.category',
        }),
      { wrapper }
    );

    expect(result?.current).toMatchSnapshot();
  });

  it('should render with extra options - breakdownField', () => {
    const { result } = renderHook(
      () =>
        useLensAttributes({
          extraOptions: { breakdownField: 'agent.type' },
          getLensAttributes: getAlertsTableLensAttributes,
          stackByField: 'event.category',
        }),
      { wrapper }
    );

    expect(result?.current).toMatchSnapshot();
  });
});
