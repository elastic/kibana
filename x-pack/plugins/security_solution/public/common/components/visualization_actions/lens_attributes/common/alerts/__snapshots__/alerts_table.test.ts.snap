// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getAlertsTableLensAttributes should render with extra options - breakdownField 1`] = `
Object {
  "description": "",
  "references": Array [
    Object {
      "id": "security-solution-my-test",
      "name": "indexpattern-datasource-layer-4aa7cf71-cf20-4e62-8ca6-ca6be6b0988b",
      "type": "index-pattern",
    },
  ],
  "state": Object {
    "adHocDataViews": Object {},
    "datasourceStates": Object {
      "formBased": Object {
        "layers": Object {
          "4aa7cf71-cf20-4e62-8ca6-ca6be6b0988b": Object {
            "columnOrder": Array [
              "2881fedd-54b7-42ba-8c97-5175dec86166",
              "75ce269b-ee9c-4c7d-a14e-9226ba0fe059",
              "f04a71a3-399f-4d32-9efc-8a005e989991",
            ],
            "columns": Object {
              "2881fedd-54b7-42ba-8c97-5175dec86166": Object {
                "dataType": "string",
                "isBucketed": true,
                "label": "Top values of event.category",
                "operationType": "terms",
                "params": Object {
                  "exclude": Array [],
                  "excludeIsRegex": false,
                  "include": Array [],
                  "includeIsRegex": false,
                  "missingBucket": false,
                  "orderBy": Object {
                    "columnId": "f04a71a3-399f-4d32-9efc-8a005e989991",
                    "type": "column",
                  },
                  "orderDirection": "desc",
                  "otherBucket": true,
                  "parentFormat": Object {
                    "id": "terms",
                  },
                  "size": 1000,
                },
                "scale": "ordinal",
                "sourceField": "event.category",
              },
              "75ce269b-ee9c-4c7d-a14e-9226ba0fe059": Object {
                "dataType": "string",
                "isBucketed": true,
                "label": "Top values of agent.type",
                "operationType": "terms",
                "params": Object {
                  "exclude": Array [],
                  "excludeIsRegex": false,
                  "include": Array [],
                  "includeIsRegex": false,
                  "missingBucket": false,
                  "orderBy": Object {
                    "columnId": "f04a71a3-399f-4d32-9efc-8a005e989991",
                    "type": "column",
                  },
                  "orderDirection": "desc",
                  "otherBucket": true,
                  "parentFormat": Object {
                    "id": "terms",
                  },
                  "size": 1000,
                },
                "scale": "ordinal",
                "sourceField": "agent.type",
              },
              "f04a71a3-399f-4d32-9efc-8a005e989991": Object {
                "dataType": "number",
                "isBucketed": false,
                "label": "Count of agent.type",
                "operationType": "count",
                "params": Object {
                  "emptyAsNull": true,
                },
                "scale": "ratio",
                "sourceField": "agent.type",
              },
            },
            "incompleteColumns": Object {},
            "sampling": 1,
          },
        },
      },
      "textBased": Object {
        "layers": Object {},
      },
    },
    "filters": Array [
      Object {
        "meta": Object {
          "alias": null,
          "disabled": false,
          "key": "_index",
          "negate": false,
          "params": Array [
            "signal-index",
          ],
          "type": "phrases",
        },
        "query": Object {
          "bool": Object {
            "minimum_should_match": 1,
            "should": Array [
              Object {
                "match_phrase": Object {
                  "_index": "signal-index",
                },
              },
            ],
          },
        },
      },
      Object {
        "meta": Object {
          "alias": null,
          "disabled": false,
          "key": "host.id",
          "negate": false,
          "params": Object {
            "query": "123",
          },
          "type": "phrase",
        },
        "query": Object {
          "match_phrase": Object {
            "host.id": "123",
          },
        },
      },
    ],
    "internalReferences": Array [],
    "query": Object {
      "language": "kql",
      "query": "host.name: *",
    },
    "visualization": Object {
      "columns": Array [
        Object {
          "columnId": "2881fedd-54b7-42ba-8c97-5175dec86166",
          "isTransposed": false,
          "width": 362,
        },
        Object {
          "columnId": "f04a71a3-399f-4d32-9efc-8a005e989991",
          "isTransposed": false,
        },
        Object {
          "columnId": "75ce269b-ee9c-4c7d-a14e-9226ba0fe059",
          "isTransposed": false,
        },
      ],
      "layerId": "4aa7cf71-cf20-4e62-8ca6-ca6be6b0988b",
      "layerType": "data",
    },
  },
  "title": "Alerts",
  "visualizationType": "lnsDatatable",
}
`;

exports[`getAlertsTableLensAttributes should render with extra options - filters 1`] = `
Object {
  "description": "",
  "references": Array [
    Object {
      "id": "security-solution-my-test",
      "name": "indexpattern-datasource-layer-4aa7cf71-cf20-4e62-8ca6-ca6be6b0988b",
      "type": "index-pattern",
    },
  ],
  "state": Object {
    "adHocDataViews": Object {},
    "datasourceStates": Object {
      "formBased": Object {
        "layers": Object {
          "4aa7cf71-cf20-4e62-8ca6-ca6be6b0988b": Object {
            "columnOrder": Array [
              "2881fedd-54b7-42ba-8c97-5175dec86166",
              "75ce269b-ee9c-4c7d-a14e-9226ba0fe059",
              "f04a71a3-399f-4d32-9efc-8a005e989991",
            ],
            "columns": Object {
              "2881fedd-54b7-42ba-8c97-5175dec86166": Object {
                "dataType": "string",
                "isBucketed": true,
                "label": "Top values of event.category",
                "operationType": "terms",
                "params": Object {
                  "exclude": Array [],
                  "excludeIsRegex": false,
                  "include": Array [],
                  "includeIsRegex": false,
                  "missingBucket": false,
                  "orderBy": Object {
                    "columnId": "f04a71a3-399f-4d32-9efc-8a005e989991",
                    "type": "column",
                  },
                  "orderDirection": "desc",
                  "otherBucket": true,
                  "parentFormat": Object {
                    "id": "terms",
                  },
                  "size": 1000,
                },
                "scale": "ordinal",
                "sourceField": "event.category",
              },
              "75ce269b-ee9c-4c7d-a14e-9226ba0fe059": Object {
                "dataType": "string",
                "isBucketed": true,
                "label": "Top values of undefined",
                "operationType": "terms",
                "params": Object {
                  "exclude": Array [],
                  "excludeIsRegex": false,
                  "include": Array [],
                  "includeIsRegex": false,
                  "missingBucket": false,
                  "orderBy": Object {
                    "columnId": "f04a71a3-399f-4d32-9efc-8a005e989991",
                    "type": "column",
                  },
                  "orderDirection": "desc",
                  "otherBucket": true,
                  "parentFormat": Object {
                    "id": "terms",
                  },
                  "size": 1000,
                },
                "scale": "ordinal",
                "sourceField": undefined,
              },
              "f04a71a3-399f-4d32-9efc-8a005e989991": Object {
                "dataType": "number",
                "isBucketed": false,
                "label": "Count of undefined",
                "operationType": "count",
                "params": Object {
                  "emptyAsNull": true,
                },
                "scale": "ratio",
                "sourceField": undefined,
              },
            },
            "incompleteColumns": Object {},
            "sampling": 1,
          },
        },
      },
      "textBased": Object {
        "layers": Object {},
      },
    },
    "filters": Array [
      Object {
        "meta": Object {
          "alias": null,
          "disabled": false,
          "key": "_index",
          "negate": false,
          "params": Array [
            ".alerts-security.alerts-default",
          ],
          "type": "phrases",
        },
        "query": Object {
          "bool": Object {
            "minimum_should_match": 1,
            "should": Array [
              Object {
                "match_phrase": Object {
                  "_index": ".alerts-security.alerts-default",
                },
              },
            ],
          },
        },
      },
      Object {
        "meta": Object {
          "alias": null,
          "disabled": false,
          "key": "_index",
          "negate": false,
          "params": Array [
            "signal-index",
          ],
          "type": "phrases",
        },
        "query": Object {
          "bool": Object {
            "minimum_should_match": 1,
            "should": Array [
              Object {
                "match_phrase": Object {
                  "_index": "signal-index",
                },
              },
            ],
          },
        },
      },
      Object {
        "meta": Object {
          "alias": null,
          "disabled": false,
          "key": "host.id",
          "negate": false,
          "params": Object {
            "query": "123",
          },
          "type": "phrase",
        },
        "query": Object {
          "match_phrase": Object {
            "host.id": "123",
          },
        },
      },
    ],
    "internalReferences": Array [],
    "query": Object {
      "language": "kql",
      "query": "host.name: *",
    },
    "visualization": Object {
      "columns": Array [
        Object {
          "columnId": "2881fedd-54b7-42ba-8c97-5175dec86166",
          "isTransposed": false,
          "width": 362,
        },
        Object {
          "columnId": "f04a71a3-399f-4d32-9efc-8a005e989991",
          "isTransposed": false,
        },
        Object {
          "columnId": "75ce269b-ee9c-4c7d-a14e-9226ba0fe059",
          "isTransposed": false,
        },
      ],
      "layerId": "4aa7cf71-cf20-4e62-8ca6-ca6be6b0988b",
      "layerType": "data",
    },
  },
  "title": "Alerts",
  "visualizationType": "lnsDatatable",
}
`;

exports[`getAlertsTableLensAttributes should render without extra options 1`] = `
Object {
  "description": "",
  "references": Array [
    Object {
      "id": "security-solution-my-test",
      "name": "indexpattern-datasource-layer-4aa7cf71-cf20-4e62-8ca6-ca6be6b0988b",
      "type": "index-pattern",
    },
  ],
  "state": Object {
    "adHocDataViews": Object {},
    "datasourceStates": Object {
      "formBased": Object {
        "layers": Object {
          "4aa7cf71-cf20-4e62-8ca6-ca6be6b0988b": Object {
            "columnOrder": Array [
              "2881fedd-54b7-42ba-8c97-5175dec86166",
              "75ce269b-ee9c-4c7d-a14e-9226ba0fe059",
              "f04a71a3-399f-4d32-9efc-8a005e989991",
            ],
            "columns": Object {
              "2881fedd-54b7-42ba-8c97-5175dec86166": Object {
                "dataType": "string",
                "isBucketed": true,
                "label": "Top values of event.category",
                "operationType": "terms",
                "params": Object {
                  "exclude": Array [],
                  "excludeIsRegex": false,
                  "include": Array [],
                  "includeIsRegex": false,
                  "missingBucket": false,
                  "orderBy": Object {
                    "columnId": "f04a71a3-399f-4d32-9efc-8a005e989991",
                    "type": "column",
                  },
                  "orderDirection": "desc",
                  "otherBucket": true,
                  "parentFormat": Object {
                    "id": "terms",
                  },
                  "size": 1000,
                },
                "scale": "ordinal",
                "sourceField": "event.category",
              },
              "75ce269b-ee9c-4c7d-a14e-9226ba0fe059": Object {
                "dataType": "string",
                "isBucketed": true,
                "label": "Top values of undefined",
                "operationType": "terms",
                "params": Object {
                  "exclude": Array [],
                  "excludeIsRegex": false,
                  "include": Array [],
                  "includeIsRegex": false,
                  "missingBucket": false,
                  "orderBy": Object {
                    "columnId": "f04a71a3-399f-4d32-9efc-8a005e989991",
                    "type": "column",
                  },
                  "orderDirection": "desc",
                  "otherBucket": true,
                  "parentFormat": Object {
                    "id": "terms",
                  },
                  "size": 1000,
                },
                "scale": "ordinal",
                "sourceField": undefined,
              },
              "f04a71a3-399f-4d32-9efc-8a005e989991": Object {
                "dataType": "number",
                "isBucketed": false,
                "label": "Count of undefined",
                "operationType": "count",
                "params": Object {
                  "emptyAsNull": true,
                },
                "scale": "ratio",
                "sourceField": undefined,
              },
            },
            "incompleteColumns": Object {},
            "sampling": 1,
          },
        },
      },
      "textBased": Object {
        "layers": Object {},
      },
    },
    "filters": Array [
      Object {
        "meta": Object {
          "alias": null,
          "disabled": false,
          "key": "_index",
          "negate": false,
          "params": Array [
            "signal-index",
          ],
          "type": "phrases",
        },
        "query": Object {
          "bool": Object {
            "minimum_should_match": 1,
            "should": Array [
              Object {
                "match_phrase": Object {
                  "_index": "signal-index",
                },
              },
            ],
          },
        },
      },
      Object {
        "meta": Object {
          "alias": null,
          "disabled": false,
          "key": "host.id",
          "negate": false,
          "params": Object {
            "query": "123",
          },
          "type": "phrase",
        },
        "query": Object {
          "match_phrase": Object {
            "host.id": "123",
          },
        },
      },
    ],
    "internalReferences": Array [],
    "query": Object {
      "language": "kql",
      "query": "host.name: *",
    },
    "visualization": Object {
      "columns": Array [
        Object {
          "columnId": "2881fedd-54b7-42ba-8c97-5175dec86166",
          "isTransposed": false,
          "width": 362,
        },
        Object {
          "columnId": "f04a71a3-399f-4d32-9efc-8a005e989991",
          "isTransposed": false,
        },
        Object {
          "columnId": "75ce269b-ee9c-4c7d-a14e-9226ba0fe059",
          "isTransposed": false,
        },
      ],
      "layerId": "4aa7cf71-cf20-4e62-8ca6-ca6be6b0988b",
      "layerType": "data",
    },
  },
  "title": "Alerts",
  "visualizationType": "lnsDatatable",
}
`;
